doc/architecture/backend.md

```diff
@@ -0,0 +1,60 @@
+# Backend Architecture
+The user always first selects a paper in the frontend and then filters and sorts the paper family. 
+For this, the backend should compute the family of a paper and prepare the output so that afterwards the backend can 
+filter quickly and the frontend can sort quickly. 
+
+## Backend Components
+The backend consists of a Flask controller, a service, and a repository. 
+The Flask controller manages the REST API for the frontend. 
+The service takes over the actual backend logic. 
+We use the repository to access the database entities. 
+
+![../img/img_06.png](../img/img_15.png)
+
+### Repository
+The getPapers call of the repository recursively queries the database for the family of the paper.
+Query: 
+                
+                with recursive family(from_paper, from_title, from_abstract, from_year, to_paper, to_title, to_abstract, to_year) as (
+                	select pf.id, pf.title, pf.abstract, pf.year, pt.id, pt.title, pt.abstract, pt.year
+                    from paper pf, reference r, paper pt
+                    where pf.id == r.from_paper and pf.title == '" + title + "' and pt.id == r.to_paper
+                            
+                    	UNION ALL
+                  
+                    select f.to_paper as from_paper, f.to_title as from_title, f.to_abstract as from_abstract, f.to_year as from_year, pt.id as to_paper, pt.title as to_title, pt.abstract as to_abstract, pt.year as to_year
+                    from family f, reference r, paper pt
+                    where f.to_paper == r.from_paper and pt.id == r.to_paper)
```

Comment:
Usually, SQL uses only a single `=` for comparisons. But I guess we already discussed this offline